### PR TITLE
Create ui.yml

### PR DESCRIPTION
**Pull request by gynt**
This is a draft for an ui.yml
Let me know what you think.
It also raises new ideas on how to define the ui.yml
I am for example not happy with how min and max are not a subdeclaration of value.
Also, choice boxes and number uis don't work nicely yet with columns setting of GroupBox > 1

### CODE REVIEW

**Comment by TheRedDaemon** [comment]
Looks good.
Putting everything value related under value might definitly help with overview.

How should the localization be done?
Do you already have a system in mind?

It might be possible to load it into the general i18n system. Or do you have some custom solution?

(Changed the target branch btw. Main is for the stuff that is intended to be in the UCP3 shortly.)

**Comment by gynt** [comment]
Can i18n resolve at runtime? I am currently doing a system where localisation is provided via a locales folder, and then the `{{ key }}` syntax can be used to do locales. I suppose if we write `{{ graphicsApiReplacer.window.type.description }}` it could resolve to the extension locale folder?

**Comment by TheRedDaemon** [comment]
It can load new translations during runtime, that is possible. Of course it would need a few adjustments, but it should be possible to load the module localizations as new namespaces. I would say, that there should only be keys given, that not include the module name. The ui config just provides the keys, which will be all be part of the modules namespace and the namespace is the module name. Just to mention this, the new system has a bit of a trade off: If localization is done and kept per module, our translators would need to add to a potential increasing amount of repos. But that should be worth it.

File: ui.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+- display: GroupBox
+  text: Replace the Graphics API. Allows for recording streams with software such as OBS.
+  columns: 1
+  category: [Graphics API Replacer]
+  children:
+  - display: GroupBox
+    header: Display settings
+    text: Set the display settings, such as windowed mode
+    columns: 2
```
Comment (gynt) [inline ui.yml:9]: columns: 1 is better looking

File: ui.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    category: []
+    children:
+    - url: graphicsApiReplacer.window.type
+      text: Set the window type
+      category: [Graphics API Replacer]
+      type: choice
```
Comment (gynt) [inline ui.yml:15]: `type: choice` should be moved to the `contents` keyword.

File: ui.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      display: Choice
+      choices: [window, borderlessWindow, borderlessFullscreen]
+      value:
```
Comment (gynt) [inline ui.yml:18]: `value` should be a subkeyword of `contents`

File: ui.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        default: window
```
Comment (gynt) [inline ui.yml:19]: `default` should be `value`

So:
```yaml
display: Choice
content:
  type: choice # or also 'choice' ?
  choices: [...]
  value:
    window
```

**Review by gynt — COMMENTED** [review]
these are the new elments

**Pull request merged**